### DISCUSSION TITLE
SSL Error on the

### DISCUSSION THREAD

**Discussion by munna-sysadmin** [postalserver/postal#1528] [Help with using Postal]
Hi,
I have installed postal on Ubuntu 18.04 & 20.04 by following the video, all the packages installed properly and postal service also running OK.
but all browsers shows the same error.
![Postal-Error](https://user-images.githubusercontent.com/14048789/128259634-9e6b8b9e-2d57-40c5-832c-d165d32891ac.PNG)

**Comment by adamcooke** [comment]
This doesn't look like a Postal problem. You'll need to look the config for your web proxy.

**Reply by munna-sysadmin** [reply]
hi @adamcooke,

Thanks for your answer, I am sharing all the configuration with you. and on my browser or network i am not using any web proxy. it is a dierct pc to postal server testing.

1. /opt/postal/config/postal.yml

general:
  # This can be changed to allow messages to be sent from multiple IP addresses
  use_ip_pools: false

web:
  # The host that the management interface will be available on
  host: postal.mydomain.com
  # The protocol that requests to the management interface should happen on
  protocol: https

web_server:
  # Specify configuration for the Postal web server
  bind_address: 127.0.0.1
  port: 5000

smtp_server:
  # Specify configuration to the Postal SMTP server
  port: 25

logging:
  # Specify options for the logging
  stdout: true

main_db:
  # Specify the connection details for your MySQL database
  host: 127.0.0.1
  username: root
  password: QhjN66Qv
  database: postal

message_db:
  # Specify the connection details for your MySQL server that will be house the
  # message databases for mail servers.
  host: 127.0.0.1
  username: root
  password: QhjN66Qv
  prefix: postal

rabbitmq:
  # Specify connection details for your RabbitMQ server
  host: 127.0.0.1
  username: postal
  password: QhjN66Qv
  vhost: postal

2. /opt/postal/config/Caddyfile

# Update this to match the hostname you wish to use for accessing the
# Postal web interface.
postal.mydomain.com {
  reverse_proxy 127.0.0.1:5000
}

3. postal status

root@postal:~# postal status
      Name                     Command               State   Ports
------------------------------------------------------------------
postal_cron_1       /docker-entrypoint.sh post ...   Up
postal_requeuer_1   /docker-entrypoint.sh post ...   Up
postal_smtp_1       /docker-entrypoint.sh post ...   Up
postal_web_1        /docker-entrypoint.sh post ...   Up
postal_worker_1     /docker-entrypoint.sh post ...   Up

4. docker ps

CONTAINER ID   IMAGE                               COMMAND                  CREATED              STATUS              PORTS                                                                      NAMES
c8a747e33abe   ghcr.io/postalserver/postal:2.0.0   "/docker-entrypoint.…"   About a minute ago   Up About a minute                                                                              postal_cron_1
3545ac2b1207   ghcr.io/postalserver/postal:2.0.0   "/docker-entrypoint.…"   About a minute ago   Up About a minute                                                                              postal_requeuer_1
95cd9d3908ab   ghcr.io/postalserver/postal:2.0.0   "/docker-entrypoint.…"   About a minute ago   Up About a minute                                                                              postal_worker_1
43a440a8fe90   ghcr.io/postalserver/postal:2.0.0   "/docker-entrypoint.…"   About a minute ago   Up About a minute                                                                              postal_web_1
1cb663502fdd   ghcr.io/postalserver/postal:2.0.0   "/docker-entrypoint.…"   About a minute ago   Up About a minute                                                                              postal_smtp_1
3b226d316820   caddy                               "caddy run --config …"   About a minute ago   Up About a minute                                                                              postal-caddy
5815fa7b0f85   rabbitmq:3.8                        "docker-entrypoint.s…"   8 hours ago          Up 8 hours          4369/tcp, 5671/tcp, 15691-15692/tcp, 25672/tcp, 127.0.0.1:5672->5672/tcp   postal-rabbitmq
56628e3027c5   mariadb                             "docker-entrypoint.s…"   8 hours ago          Up 8 hours          127.0.0.1:3306->3306/tcp                                                   postal-mariadb



Thanks

**Reply by munna-sysadmin** [reply]
hi @adamcooke 

I have configured the postal without SSL with only 80 port and it is working just fine. web interface just comes up.
When traffic coming through Caddy 443 it is causing the issue, please help.

**Reply by adamcooke** [reply]
Did you change the port the postal web server was listening on?

**Reply by munna-sysadmin** [reply]
Yes, i made it to listen on port 80, without caddy. It was working fine.

**Reply by munna-sysadmin** [reply]
Hi @adamcooke 

My problem is solved, the problem was local dns resolution and on ubuntu hostname file should contain hostname as fqdn, now caddy & ssl is working just fine.

**Reply by munna-sysadmin** [reply]
Hi @adamcooke 

Another question,  how postal is receiving local destined mail, i dont see any local user mailbox creation option. So i can use postal for email receiving and delivering to another server, just like an email router, right?

**Reply by willpower232** [reply]
Postal does not hold on to email at all (no local user mailboxes), there needs to be a route set up to tell it where emails should go otherwise not much will happen.

**Reply by munna-sysadmin** [reply]
@willpower232 
Thanks for the clarification.

**Reply by hswamy** [reply]
Hello @munna-sysadmin , I have similar error with SSL secure connections. Can you please let me know how did you fix this ?

Your help much appreciated

**Reply by kayastha1988** [reply]
I am having some kind issue like below:

web_1       | Puma starting in single mode...
web_1       | * Version 4.3.8 (ruby 2.6.8-p205), codename: Mysterious Traveller
web_1       | * Min threads: 5, max threads: 5
web_1       | * Environment: production
web_1       | Errno::EADDRNOTAVAIL: Cannot assign requested address - bind(2) for "172.105.33.83" port 5000

How can this be solved?
Looking forward to hear. 
Thanks

**Reply by willpower232** [reply]
@kayastha1988 does that IP address appear if you run `ip a`?

**Reply by stevebrainng** [reply]
I Similar Issue when I check log I check this
web_1       | Puma starting in single mode...
web_1       | * Version 4.3.10 (ruby 2.6.8-p205), codename: Mysterious Traveller
web_1       | * Min threads: 5, max threads: 5
web_1       | * Environment: production
web_1       | Errno::EADDRNOTAVAIL: Cannot assign requested address - bind(2) for "XX.XX.XX.X." port 5000
web_1       |   /usr/local/bundle/gems/puma-4.3.10/lib/puma/binder.rb:229:in `initialize'
web_1       |   /usr/local/bundle/gems/puma-4.3.10/lib/puma/binder.rb:229:in `new'
web_1       |   /usr/local/bundle/gems/puma-4.3.10/lib/puma/binder.rb:229:in `add_tcp_listener'
web_1       |   /usr/local/bundle/gems/puma-4.3.10/lib/puma/binder.rb:106:in `block in parse'
web_1       |   /usr/local/bundle/gems/puma-4.3.10/lib/puma/binder.rb:90:in `each'
web_1       |   /usr/local/bundle/gems/puma-4.3.10/lib/puma/binder.rb:90:in `parse'
web_1       |   /usr/local/bundle/gems/puma-4.3.10/lib/puma/runner.rb:161:in `load_and_bind'
web_1       |   /usr/local/bundle/gems/puma-4.3.10/lib/puma/single.rb:98:in `run'
web_1       |   /usr/local/bundle/gems/puma-4.3.10/lib/puma/launcher.rb:172:in `run'
web_1       |   /usr/local/bundle/gems/puma-4.3.10/lib/puma/cli.rb:80:in `run'
web_1       |   /usr/local/bundle/gems/puma-4.3.10/bin/puma:10:in `<top (required)>'
web_1       |   /usr/local/bundle/bin/puma:23:in `load'
web_1       |   /usr/local/bundle/bin/puma:23:in `<top (required)>'
web_1       | bundler: failed to load command: puma (/usr/local/bundle/bin/puma)


when I run "ip a" to check the IP Address it's not showing up among the result printed by "IP a"


And also when I run "postal status" Command I get 

      Name                     Command               State    Ports
-------------------------------------------------------------------
postal_cron_1       /docker-entrypoint.sh post ...   Up
postal_requeuer_1   /docker-entrypoint.sh post ...   Up
postal_smtp_1       /docker-entrypoint.sh post ...   Up
postal_web_1        /docker-entrypoint.sh post ...   Exit 1
postal_worker_1     /docker-entrypoint.sh post ...   Exit 1

**Comment by willpower232** [comment]
@stevebrainng this means that either the IP address you have entered in the config file is not available to the server or something is already on that port

> when I run "ip a" to check the IP Address it's not showing up among the result printed by "IP a"

You have to use an IP address shown by `ip a` as those are the ones which are available to your server. It may be that your server provider does some mapping so that you cannot see the real external IP address from inside the server.
